src/sys/wait.rs

```diff
@@ -235,6 +235,57 @@ pub fn waitpid<P: Into<Option<Pid>>>(pid: P, options: Option<WaitPidFlag>) -> Re
     }
 }
 
+
+/// The same as calling waitpid(None, None)
 pub fn wait() -> Result<WaitStatus> {
     waitpid(None, None)
 }
+
+
+/// Returns an iterator over all currently available events of child processes.
+/// 
+/// This iterator is best used with a signal handler on the signal SIGCHLD
+/// (Note that this example uses the signal_hook crate. This is just for clarity, it would of course work with any other mechanism to
+/// receive signals)
+/// ```ignore
+/// for signal in signals.forever() {
```

Comment:
The example isn't very illustrative since it relies on an external crate.  But nix includes signal handling functionality.  Could you rework the example to eliminate the external crate?

Reply:
This is a bit tough, since handling signals is a bit tricky if there is actual work to be done. i think this is too elaborated to have as an example. I removed the external crate but I added some more text to point this out to users.

Does this count as the doc test you mentioned above? :)

Reply:
Well, no, because it doesn't do anything.  It doesn't create any child processes, and never gets SIGCHLD.  What I would do if I were you would be to set a signal handler, then fork the child processes, then poll until the signal is received.  That's the simplest way to handle signals.  For an example, see `test_write_sigev_signal` in tests/sys/test_aio.rs .

Reply:
I went a bit of another route for the test since the signal is not actually needed and makes the test more complex than I think it needs to be. I hope this is acceptable regardless. If you think that signal handling should be part of this I will add it of course.